dash-spv-apple-bindings/src/coinjoin.rs

```diff
@@ -19,80 +21,102 @@ use dash_spv_masternode_processor::ffi::unboxer::unbox_any;
+pub unsafe extern "C" fn register_wallet_ex(
+    context: *const c_void,
     options_ptr: *mut CoinJoinClientOptions,
     get_wallet_transaction: GetWalletTransaction,
-    destroy_wallet_transaction: DestroyWalletTransaction,
+    sign_transaction: SignTransaction,
+    destroy_transaction: DestroyWalletTransaction,
     is_mine: IsMineInput,
+    commit_transaction: CommitTransaction,
+    masternode_by_hash: MasternodeByHash,
+    destroy_masternode: DestroyMasternode,
+    valid_mns_count: ValidMasternodeCount,
+    is_synced: IsBlockchainSynced,
+    fresh_coinjoin_key: FreshCoinJoinAddress,
+    count_inputs_with_amount: InputsWithAmount,
     available_coins: AvailableCoins,
     destroy_gathered_outputs: DestroyGatheredOutputs,
     selected_coins: SelectCoinsGroupedByAddresses,
     destroy_selected_coins: DestroySelectedCoins,
-    sign_transaction: SignTransaction,
-    count_inputs_with_amount: InputsWithAmount,
-    fresh_coinjoin_key: FreshCoinJoinAddress,
-    commit_transaction: CommitTransaction,
-    masternode_by_hash: MasternodeByHash,
-    destroy_masternode: DestroyMasternode,
-    context: *const c_void
-) -> *mut CoinJoinClientSession {
-    let session = CoinJoinClientSession::new(
-        std::ptr::read(coinjoin_ptr),
+    is_masternode_or_disconnect_requested: IsMasternodeOrDisconnectRequested
+) -> *mut WalletEx {
+    let wallet_ex =  WalletEx::new(
+        context, 
         std::ptr::read(options_ptr), 
-        sign_transaction,
         get_wallet_transaction, 
-        destroy_wallet_transaction, 
+        sign_transaction,
+        destroy_transaction, 
         is_mine, 
-        available_coins,
+        available_coins, 
         destroy_gathered_outputs,
         selected_coins, 
         destroy_selected_coins,
         count_inputs_with_amount,
         fresh_coinjoin_key,
         commit_transaction,
-        // masternode_by_hash, TODO
-        // destroy_masternode,
+        masternode_by_hash, 
+        destroy_masternode, 
+        valid_mns_count, 
+        is_synced,
+        is_masternode_or_disconnect_requested
+    );
+
+    println!("[RUST] CoinJoin: register_wallet_ex");
+    let wallet_ex_rc = Rc::new(RefCell::new(wallet_ex));
+    return Rc::into_raw(wallet_ex_rc) as *mut WalletEx; // TODO: check refs count before dropping
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn register_client_manager(
+    context: *const c_void,
+    wallet_ex_ptr: *mut WalletEx,
+    options_ptr: *mut CoinJoinClientOptions,
+    get_masternode_list: GetMasternodeList,
+    destroy_mn_list: DestroyMasternodeList,
+    get_input_value_by_prevout_hash: GetInputValueByPrevoutHash,
+    has_chain_lock: HasChainLock,
+    destroy_input_value: DestroyInputValue
+) -> *mut CoinJoinClientManager {
+    let wallet_ex_rc = Rc::from_raw(wallet_ex_ptr as *mut RefCell<WalletEx>);
+    // Increment the strong count to avoid deallocation when Rc goes out of scope here
+    std::mem::forget(wallet_ex_rc.clone());
```

Comment:
Some ChatGPT assisted code to be able to keep the WalletEx  reference in Obj-C and pass it around where its needed. Will need to re-check for leaks and ref count later.